### PR TITLE
Scoreboard

### PR DESCRIPTION
**Pull request by ZombieAlienRobot**
## Description of changes
Gets the scoreboard when queried. The scoreboard is sorted by power score from highest to lowest.

  
## How has this been tested?
Please describe the test strategy you followed.

- [X] automated unit test
- [X] automated integration test
- [ ] automated acceptance test
- [X] manual, exploratory test
    
## Checklist before requesting a review
- [X] My code follows the [coding guidelines](https://github.com/IT-REX-Platform/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of this project
- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] My code fulfills all acceptance criteria
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] The test coverage (line and branch) is reasonably high, especially on Service classes and other classes with complex logic
- [X] I have made corresponding changes to the documentation
- [X] I have added explanation of architectural decision and rationales to [wiki/adr](https://github.com/IT-REX-Platform/wiki/tree/main/adr)
- [X] I have updated the changes in the ticket description

## Checklist for reviewer
- [ ] The code works and does not throw errors
- [ ] The code is easy to understand and there are no confusing parts
- [ ] The code follows the [coding guidelines](https://github.com/IT-REX-Platform/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of this project
- [ ] The code change accomplishes what it is supposed to do
- [ ] I cannot think of any use case in which the code does not behave as intended
- [ ] The added and existing tests reasonably cover the code change
- [ ] I cannot think of any test cases, input or edge cases that should be tested in addition
- [ ] Description of the change is included in the documentation

### CODE REVIEW

File: src/main/java/de/unistuttgart/iste/gits/reward/persistence/dao/AllRewardScoresEntity.java [hunk 1]
```diff
@@ -19,22 +21,27 @@ public class AllRewardScoresEntity {
...
+    @OnDelete(action = OnDeleteAction.CASCADE)
     private RewardScoreEntity fitness;
     @OneToOne(optional = false, cascade = CascadeType.ALL)
+    @OnDelete(action = OnDeleteAction.CASCADE)
     private RewardScoreEntity growth;
     @OneToOne(optional = false, cascade = CascadeType.ALL)
+    @OnDelete(action = OnDeleteAction.CASCADE)
     private RewardScoreEntity strength;
     @OneToOne(optional = false, cascade = CascadeType.ALL)
+    @OnDelete(action = OnDeleteAction.CASCADE)
```
Comment (PaulBredl) [inline src/main/java/de/unistuttgart/iste/gits/reward/persistence/dao/AllRewardScoresEntity.java:36]: ah okay, braucht man das zusätzlich zu cascade = CascadeType.ALL? 

dann ist das vermutlich die ursache für den content service bug.

Comment (ZombieAlienRobot) [reply src/main/java/de/unistuttgart/iste/gits/reward/persistence/dao/AllRewardScoresEntity.java:36]: Zumindest sind sonst die Query tests gefailed

File: src/test/java/de/unistuttgart/iste/gits/reward/service/RewardServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.Mockito.*;
+
+public class RewardServiceTest {
```
Comment (PaulBredl) [inline src/test/java/de/unistuttgart/iste/gits/reward/service/RewardServiceTest.java:19]: Meiner Meinung nach ist das sogar overkill, wenn der api test läuft impliziert das ja eingentlich, dass die service klasse funktioniert. Aber schaden tut es nicht 👍

File: src/main/java/de/unistuttgart/iste/gits/reward/persistence/dao/AllRewardScoresEntity.java (hunk 1 above)
Comment (myluki2000) [reply src/main/java/de/unistuttgart/iste/gits/reward/persistence/dao/AllRewardScoresEntity.java:36]: aber mit query sollte das doch nichts zu tun haben oder?`Das ist ja nur für den Fall dass gelöscht wird.

File: src/main/java/de/unistuttgart/iste/gits/reward/service/RewardService.java [hunk 3]
```diff
@@ -144,4 +143,23 @@ public RewardScoreEntity initializeRewardScoreEntity(int initialValue) {
...
     }
 
+
+    /**
+     * Returns the scoreboard for a specific course. Sorted by power from highest to lowest.
+     * @param courseId of the course for which the scoreboard should be retrieved
+     * @return scoreboard
+     */
+    public List<ScoreboardItem> getScoreboard(UUID courseId) {
+        List<AllRewardScoresEntity> allRewardScoresEntity = rewardScoresRepository.findAllRewardScoresEntitiesById_CourseId(courseId);
```
Comment (PaulBredl) [inline src/main/java/de/unistuttgart/iste/gits/reward/service/RewardService.java:153]: naming ist bisschen verwirrend, weil du ja ne liste an allRewardScoresEntities hast, aber das ist ein nitpick

**Review by myluki2000 — APPROVED** [review]

File: src/main/java/de/unistuttgart/iste/gits/reward/service/RewardService.java (hunk 3 above)
Comment (ZombieAlienRobot) [reply src/main/java/de/unistuttgart/iste/gits/reward/service/RewardService.java:153]: True, ich ändere es schnell

File: src/main/java/de/unistuttgart/iste/gits/reward/persistence/dao/AllRewardScoresEntity.java (hunk 1 above)
Comment (PaulBredl) [reply src/main/java/de/unistuttgart/iste/gits/reward/persistence/dao/AllRewardScoresEntity.java:36]: wahrscheinlich das clean up nach dem test ist gefailed
